src/common/admin_socket.cc

```diff
@@ -299,6 +299,18 @@ void AdminSocket::chown(uid_t uid, gid_t gid)
   }
 }
 
+void AdminSocket::chmod(mode_t mode)
+{
+  if (m_sock_fd >= 0) {
+    int r = ::chmod(m_path.c_str(), mode);
```

Comment:
If `fildes` refers to a socket, the behavior of fchmod() is unspecified.[1]

[1] https://linux.die.net/man/3/fchmod